### PR TITLE
1zhaohel/tri 106 Debug Overall Fairness Score Issue

### PR DESCRIPTION
**Pull request by 1zhaohel**
**Description**
Previously, different transactions with varied biases would achieve a low fairness score. For example, very biased dataset receives a F and no bias dataset receives a D-. 

Instead, using accuracy as the score seems to be more reflective of the datasets. Very biased dataset now receives a C and no bias dataset receives an A+. 

Additionally, refactored clean_dataset to use protected_attributes instead of hardcode.

### CODE REVIEW

**Comment by 1zhaohel** [comment]
Also ig netlify preview only displays frontend changes because the backend changes are not reflected in the link.

**Comment by aakshataa** [comment]
I think this accuracy score is important to include in our algorithm, but I think we should continue to use our statistical metrics since they are a core part of our program. Maybe instead, we can 'scale' our scores to reflect the bias in our test datasets. 

For example, our 'no bias' dataset should have a high score (~A). If we use this as our best score, we can scale everything enough to where this D- changes to an A. Every other datasets score should follow from that. 

Another option is to also modify the 'ranges' of fairness

**Comment by 1zhaohel** [comment]
Now No Bias gives A+, Biased gives F and Medium gives C+

**Review by aakshataa — CHANGES_REQUESTED** [review]

File: frontend/src/views/components/Dashboard/BiasScore/BiasScore.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Card, CardContent, CardHeader, Typography } from "@mui/material";
+
+function BiasScore({score}) {
+    return (
+        <Card  sx={{flex: "1"}}>
+            <CardHeader title=" Your Fairness Score"/>
+            <CardContent>
+                <Typography variant="h1">{score}</Typography>
```
[the remark this answers is not in the document]
Comment (1zhaohel) [reply frontend/src/views/components/Dashboard/BiasScore/BiasScore.tsx]: good catch!

File: backend/core/entities/bias_metrics.py [hunk 2]
```diff
@@ -96,3 +111,8 @@ def calculate_bias_metrics(self, protected_attribute) -> dict[str: float]:
             metrics['Equal Opportunity Difference'] = round(float(metric.equal_opportunity_difference()), 2)
 
         return metrics
+
+    # def get_accuracy(self) -> float:
```
[the remark this answers is not in the document]
Comment (1zhaohel) [reply backend/core/entities/bias_metrics.py:117]: I think we will use this function in the overview so it's fine to leave it in for now

**Review by 1michhu1 — APPROVED** [review]
yep, looks good to me (discussed changes and questions in person)

**Pull request merged**